pkg/controller/migration/namespace_migration.go

```diff
@@ -796,3 +806,24 @@ func (m *CoreNamespaceMigration) removeNodeLabel(ctx context.Context, nodeName,
 		return true, nil
 	})
 }
+
+// waitForOperatorCalicoNodeDaemonsetReady waits until the 'new' calico-node daemonset in
+// the calico-system namespace is ready before continuing, it will wait up to
+// 1 minutes before returning with an error.
+func (m *CoreNamespaceMigration) waitForOperatorCalicoNodeDaemonsetReady(ctx context.Context, log logr.Logger) error {
+	return wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
Is 1 minute going to be long enough for larger clusters with hundreds of nodes? I suspect it might not - perhaps we need a variable time here based on the size of the DS?

Reply:
That's an excellent question. It does not enough. But in this way, the reconcilation is not blocked for long time. And the continues reconcilation just solve the problem - it waits on this 1 minutes, and ends. And an another rounds starts.

Although I agree we can make this better. 
We have tested with 700 nodes, and it took 6hrs 30 mins total with this change (tested with Kubernetes 1.29). The change definitely increases the time, but it makes it more stable based on our testing (this change was requested from our update team). Also, the rolling update settings is important here.

Thanks!